File: app/src/main/java/com/test/xyz/demo/domain/interactor/weather/WeatherInteractor.java
package com.test.xyz.demo.domain.interactor.weather;

import com.test.xyz.demo.domain.model.weather.WeatherSummaryInfo;

import io.reactivex.Observable;

public interface WeatherInteractor {
    Observable<WeatherSummaryInfo> getWeatherInformation(String userName, String cityName);
}

File: app/src/main/kotlin/com/test/xyz/demo/domain/interactor/weather/WeatherInteractor.kt
package com.test.xyz.demo.domain.interactor.weather

import com.test.xyz.demo.domain.model.weather.WeatherSummaryInfo

import io.reactivex.Observable

interface WeatherInteractor {
    fun getWeatherInformation(userName: String, cityName: String): Observable<WeatherSummaryInfo>
}
